### PR TITLE
Adding vector functions

### PR DESCRIPTION
**Pull request by JosephNewmanTG**
Doc Ticket Link: https://graphsql.atlassian.net/browse/DOC-2191

This adds 1 page and updates to the index:
- 	modified:   modules/querying/pages/func/index.adoc
- 	modified:   modules/querying/pages/func/vector-functions.adoc

### CODE REVIEW

File: modules/querying/pages/func/vector-functions.adoc [hunk 1]
```diff
@@ -0,0 +1,1007 @@
+= Vector Functions
+:experimental:
+
+Vector Functions in TigerGraph are currently implemented as UDFs and preloaded as part of the Graph Data Science Library:
+xref:3.9@graph-ml:similarity-algorithms:index.adoc[]
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]: This is a 3.10.1 PR, but your link path points to 3.9.

File: modules/querying/pages/func/vector-functions.adoc [hunk 2]
```diff
@@ -0,0 +1,1007 @@
...
+
+
+== Vector
+=== tg_euclidean_distance_accum ()
+====  Syntax
+`tg_euclidean_distance_accum (VectorA, VectorB)`
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]: Note that it is now `tg_similarity_accum (VectorA, VectorB, "EUCLIDEAN")`

File: modules/querying/pages/func/vector-functions.adoc [hunk 3]
```diff
@@ -0,0 +1,1007 @@
...
+====
+Query::
++
+--
+[,gsql]
+----
+CREATE QUERY euclidean_example() FOR GRAPH social {
+    ListAccum<INT> @@a = [1, 2, 3];
+    ListAccum<INT> @@b = [4, 5, 6];
+    double distance = tg_euclidean_distance_accum(@@a, @@b);
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
    double distance = tg_similarity_accum(@@a, @@b, "EUCLIDEAN");
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 4]
```diff
@@ -0,0 +1,1007 @@
...
+[,json]
+----
+{
+    "jaccard_similarity": 0.5
+}
+----
+--
+====
+
+== Vector Similarity of Neighborhoods
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc:330]: Do not include this section.

File: modules/querying/pages/func/vector-functions.adoc [hunk 5]
```diff
@@ -0,0 +1,1007 @@
...
+[,json]
+----
+{
+    "distance": 5.19615
+}
+----
+--
+====
+
+== Vector Similarity
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]: This seems to an extra heading. You have 1 of the 5 above, and 4 of the 5 below. There is no reason to separate them with this heading.

File: modules/querying/pages/func/vector-functions.adoc [hunk 6]
```diff
@@ -0,0 +1,1007 @@
+= Vector Functions
+:experimental:
+
+Vector Functions in TigerGraph are currently implemented as UDFs and preloaded as part of the Graph Data Science Library:
+xref:3.9@graph-ml:similarity-algorithms:index.adoc[]
+
+
+== Vector
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]: This page is in the GSQL **Functions** documentation. It is not for **algorithms**, so do not copy the descriptions of similarity algorithms here.

My original suggestion was not to document the similarity functions here, but to direct the reader to the GDS Similarity Algorithms page, and to put the documentation there.

**Review by victorleeTG — CHANGES_REQUESTED** [review]

File: modules/querying/pages/func/vector-functions.adoc (hunk 1 above)
Comment (JosephNewmanTG) [reply modules/querying/pages/func/vector-functions.adoc]: Thank you. This has been updated.

File: modules/querying/pages/func/vector-functions.adoc (hunk 2 above)
Comment (JosephNewmanTG) [reply modules/querying/pages/func/vector-functions.adoc]: Thank you this has been updated with latest commit.

File: modules/querying/pages/func/vector-functions.adoc (hunk 3 above)
Comment (JosephNewmanTG) [reply modules/querying/pages/func/vector-functions.adoc]: This has been updated with latest commit.

File: modules/querying/pages/func/vector-functions.adoc (hunk 4 above)
Comment (JosephNewmanTG) [reply modules/querying/pages/func/vector-functions.adoc:330]: This was committed out with latest commit.

File: modules/querying/pages/func/vector-functions.adoc (hunk 5 above)
Comment (JosephNewmanTG) [reply modules/querying/pages/func/vector-functions.adoc]: Thank you. This was removed.

File: modules/querying/pages/func/vector-functions.adoc (hunk 6 above)
Comment (JosephNewmanTG) [reply modules/querying/pages/func/vector-functions.adoc]: I understand now. I will leave these here though. I don't think it is a bad idea to have just the functions listed here.

File: modules/querying/pages/func/vector-functions.adoc [hunk 7]
```diff
@@ -0,0 +1,1052 @@
+= Vector Similarity Function
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
= Vector Functions
```
This page should begin the same way that the other GSQL function pages start. We are presenting a category of functions; the category is usually defined by the type of data that is works with, in this case, vectors.  In the long run, there might be several functions on this page, so don't speak as those it's just one function.

File: modules/querying/pages/func/vector-functions.adoc [hunk 8]
```diff
@@ -0,0 +1,1052 @@
...
+:experimental:
+
+Vector Similarity Function in TigerGraph is currently implemented as UDFs and preloaded as part of the Graph Data Science Library:
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
This page lists all the functions in the GSQL query language that operate on vector data.  In general, a vector could be either a `LIST<>` type attribute or a `ListAccum<>` variable. 
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 9]
```diff
@@ -0,0 +1,1052 @@
...
+xref:graph-ml:similarity-algorithms:index.adoc[]
+
+[CAUTION]
+====
+Vector Similarity Functions are subcomponents for the Appropriate Nearest Nearest (ANN) algorithm which is still in beta.
+====
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
[IMPORTANT]
====
The current vector functions are implemented as UDFs and preloaded as part of the Graph Data Science Library and thus are not yet permanently parts of the GSQL language.
The current vector functions work with` ListAccum<>` inputs but not directly with `List<>` attributes.
====
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 10]
```diff
@@ -0,0 +1,1052 @@
...
+
+[IMPORTANT]
+====
+Beta features are for personal or R&D use and not for production use.
+====
+
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 11]
```diff
@@ -0,0 +1,1052 @@
...
+
+
+== tg_similarity_accum()
+
+The function takes 3 parameters.
+
+* The first 2 parameters are `ListAccum<>` for the two vectors being compared.
+* The 3rd parameter is a string and indicates which variation of the function to use.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
* The first two parameters are `ListAccum<>` for the two vectors being compared.
* The third parameter is a string and indicates which variation of the function to use.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 12]
```diff
@@ -0,0 +1,1052 @@
...
+[IMPORTANT]
+====
+Beta features are for personal or R&D use and not for production use.
+====
+
+
+
+== tg_similarity_accum()
+
+The function takes 3 parameters.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
Returns a similarity (or distance) measure between two vectors.
This function supports multiple similarity measures; the user selects which measure by setting one of the input parameters.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 13]
```diff
@@ -0,0 +1,1052 @@
...
+
+* The first 2 parameters are `ListAccum<>` for the two vectors being compared.
+* The 3rd parameter is a string and indicates which variation of the function to use.
++
+====
+.For example:
+
+** If the string’s value is `“COSINE”`, then the function performs cosine similarity.
+** If the string’s value is `“JACCARD”`, it performs Jaccard similarity.
+====
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
|===
| Value | Function

| "COSINE"
| cosine similarity

| "EUCLIDEAN"
| Euclidean distance

| "JACCARD"
| Jaccard similarity

| "OVERLAP"
| overlap similarity

| "PEARSON"
| Pearson correlation coefficient
|===
```

**Review by victorleeTG — CHANGES_REQUESTED** [review]

File: modules/querying/pages/func/vector-functions.adoc [hunk 14]
```diff
@@ -0,0 +1,1052 @@
...
+
+=== Euclidean Distance
+====  Syntax
+`tg_similarity_accum(VectorA, VectorB, "EUCLIDEAN")`
+
+====  Description
+The Euclidean distance between two vectors.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc:47]:
```suggestion
The Euclidean distance between two vectors.
The vectors must have the same length and have corresponding numeric elements.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 15]
```diff
@@ -0,0 +1,1052 @@
...
+----
+--
+====
+
+=== Overlap
+==== Syntax
+`tg_similarity_accum( VectorA, VectorB, "OVERLAP" )`
+
+====  Description
+The overlap coefficient between two vectors.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
The overlap coefficient between two vectors, that is, the number of items appearing in both lists, divided by the number of items in the shorter list.
Overlap similarity is intended for vectors which are unordered sets of _categorical data_, such as the sets of city names mentioned in travel blogs.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 16]
```diff
@@ -0,0 +1,1052 @@
...
+==== Return type
+`double`
+
+====  Parameters
+|===
+| Parameter | Description | Data type
+
+|`VectorA`
+|An n-dimensional vector denoted by a `+ListAccum+` of length `n`.
+|`+ListAccum<INT/UINT/FLOAT/DOUBLE>+`
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
|`+ListAccum<INT/UINT/STRING>+`
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 17]
```diff
@@ -0,0 +1,1052 @@
...
+
+|`VectorB`
+|An n-dimensional vector denoted by a `+ListAccum+` of length `n`.
+|`+ListAccum<INT/UINT/FLOAT/DOUBLE>+`
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
|`+ListAccum<INT/UINT/STRING>+`
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 18]
```diff
@@ -0,0 +1,1052 @@
...
+----
+--
+====
+
+=== Pearson
+====  Syntax
+`tg_similarity_accum(VectorA, VectorB, "PEARSON")`
+
+====  Description
+The Pearson correlation coefficient between two vectors.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc:162]:
```suggestion
The Pearson correlation coefficient between two vectors.
The vectors must have the same length and have corresponding numeric elements.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 19]
```diff
@@ -0,0 +1,1052 @@
...
+--
+====
+
+=== Cosine
+====  Syntax
+`tg_similarity_accum(VectorA, VectorB, "COSINE")`
+
+====  Description
+
+The cosine similarity between the two vectors.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc:219]:
```suggestion
The cosine similarity between the two vectors.
The vectors must have the same length and have corresponding numeric elements.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 20]
```diff
@@ -0,0 +1,1052 @@
...
+--
+====
+
+=== Jaccard
+====  Syntax
+`tg_similarity_accum(VectorA, VectorB "JACCARD")`
+
+====  Description
+
+The Jaccard similarity between the two sets.
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
The Jaccard similarity between the two sets, that is, the number of items appearing in both sets, divided by the number of unique items in the lists.
Jaccard similarity, like overlap similarity, is intended for vectors which are unordered sets of _categorical data_, such as the sets of city names mentioned in travel blogs.
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 21]
```diff
@@ -0,0 +1,1052 @@
...
+
+`double`
+
+====  Parameters
+|===
+| Parameter | Description | Data type
+
+|`VectorA`
+|An n-dimensional vector denoted by a `+ListAccum+` of length `n`.
+|`ListAccum<INT/UINT/FLOAT/DOUBLE>`
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
|`ListAccum<INT/UINT/STRING>`
```

File: modules/querying/pages/func/vector-functions.adoc [hunk 22]
```diff
@@ -0,0 +1,1052 @@
...
+
+|`VectorB`
+|An n-dimensional vector denoted by a `+ListAccum+` of length `n`.
+|`ListAccum<INT/UINT/FLOAT/DOUBLE>`
```
Comment (victorleeTG) [inline modules/querying/pages/func/vector-functions.adoc]:
```suggestion
|`ListAccum<INT/UINT/STRING>`
```

**Review by victorleeTG — CHANGES_REQUESTED** [review]

**Review by victorleeTG — APPROVED** [review]

**Pull request merged**